### ISSUE TITLE
Argocd-server API call to sync resource take a long time

### ISSUE THREAD

**Issue by dromadaire54** [argoproj/argo-cd#21127]
Checklist:
* argocd version: 2.13.1
* argocd helmchart version: 7.7.8

**Describe the bug**

From argocd UI the a simple resource like a configmap sync takes more than 4 minutes. The call directly to the argocd-server API take a long time too. The issue is for argocd app using a helmchart ecr but not the git branch target revision. Furthermore, I noticed when the argocd is completly restarted the two first call are instantly but after it takes 4min it's maybe a hint.

**Screenshots**
![image](https://github.com/user-attachments/assets/794b95df-8c0a-4520-be66-40498510d5ee)

![image](https://github.com/user-attachments/assets/350b2c5d-197f-4b7b-85fd-cc3916ead153)

**Logs**
```
2024-12-11 09:11:55.549	
time="2024-12-11T08:11:55Z" level=info msg="finished streaming call with code OK" grpc.code=OK grpc.method=Watch grpc.service=application.ApplicationService grpc.start_time="2024-12-11T08:11:42Z" grpc.time_ms=12830.461 span.kind=server system=grpc
```

**Comment by dromadaire54** [comment]
Who can give me some tips to debug this issue ? It's very annoying each synchronize from the ui takes several min.

**Comment by cliebert** [comment]
Did you notice that same day you posted this bug v2.13.2 was released. Do you have same issue with that version?

**Comment by dromadaire54** [comment]
Yes unfortunately I have the same issue. I will oblige to debug the argocd server to know where in the code the waiting time happening.

**Comment by crenshaw-dev** [comment]
I'm seeing a similar issue but with refreshes taking ~4min. So far it looks like the app controller has stale Application data and doesn't see the refresh annotation for a long time. Maybe there's a similar problem with the sync operation field.

**Comment by dromadaire54** [comment]
Yes it looks like with the sync operation. We must to know with which version the issue appeared.

**Comment by dromadaire54** [comment]
@crenshaw-dev I debug argocd server with telepresence the waiting time is in `ResolveRevision` call to get the revision. It uses repo server to retrieve the revision in the method `c.cc.Invoke`. I don't know why I delete all the netpol in the ns argocd.

**Comment by dromadaire54** [comment]
Someone can help me ?

**Comment by dromadaire54** [comment]
Hi,
@crenshaw-dev just to say the issue is fixed. It's due to the ecr repository which contained a several thousands of tags and in the repo-server gets all the tags before to execute the sync. After adding a lifecycle for the repo all the latencies for example the refresh and synchronize disappeared.
